Add powershell completion

hgraeber opened the pull request:
What does this PR change? What problem does it solve?
-----------------------------------------------------

Restic can generate completion files for bash, fish and zsh, but not for powershell.
Cobra can generate this, too. 

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

No. I have spotted that cobra can do this and wondered why restic does not use it.

Checklist
---------

- [X] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [X] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [X] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [X] I have run `gofmt` on the code in all commits.
- [X] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

rawtaz commented on doc/manual_rest.rst:
```diff
@@ -26,7 +26,7 @@ Usage help is available:
       find          Find a file, a directory or restic IDs
       forget        Remove snapshots from the repository
-      generate      Generate manual pages and auto-completion files (bash, fish, zsh)
+      generate      Generate manual pages and auto-completion files (bash, fish, zsh, powershell)
```
This entire file is generated as part of the release build process, so please remove all changes to this file :)

rawtaz commented on doc/man/restic-generate.1:
```diff
@@ -3,7 +3,7 @@
 .SH NAME
 .PP
-restic-generate - Generate manual pages and auto-completion files (bash, fish, zsh)
+restic-generate - Generate manual pages and auto-completion files (bash, fish, zsh, powershell)
```
This entire file is generated as part of the release build process, so please remove all changes to this file :)

rawtaz reviewed (changes requested):
Thanks for this! I've added a few small comments that need addressing.

rawtaz commented:
Oh, and could you please squash all commits into one and rebase it on the latest master?

hgraeber committed:
Add powershell completion

- Add code for powersehll complition available in cobra
- Add documentation for powershell completion
- Add changelog for pr3925

rawtaz reviewed (approved):
The URL in the changelog entry needs to be corrected (my bad, didn't remove the /files part), but we'll do that before the release, so no worries about that. Thanks again!
